ts/Extensions/NonCartesianSeriesZoom/NonCartesianSeriesZoom.ts

```diff
@@ -399,6 +407,21 @@ function onAfterSetChartSize(
     }
 }
 
+/**
+ * Create data labels parent group for clipping purposes after zoom-in
+ * @private
+ */
+function onDrawDataLabels(
+    this: Series
+): void {
+    if (this.hasDataLabels?.()) {
+        this.dataLabelsGroupParent ||=
+            this.chart.renderer.g()
+                .attr({ zIndex: (this.options.dataLabels as any).zIndex || 6 })
```

Comment:
This casting hides a problem of no support for dataLabels as an array I mention in the top review comment.